### PR TITLE
TMCP 1.0

### PR DESCRIPTION
**Pull request by VirxEC**
Version bump on everything
Changed readme to tell users to avoid major breaking changes via checking the major version of TMCP (all breaking changes will now be complemented by a major version increase)
Downloading the latest release for any major version 1 package
Much more flexible parsing of packets:

- Packets won't be ignored if they have dissimilar versions, only invalid versions
- `type` key under `action` is automatically capitalized
- If the `time` key in the actions `BALL`, `READY` or `DEMO` isn't present, they will be set to their default value (`-1`).
- If the `direction` key in the action `BALL` isn't present, it will be set to its default value (`[0, 0, 0]`).
- If the `direction` key in the action `BALL` has only 2 elements, a third `0` will be appended.

Missing non-optional values will still result in the packet being flagged as invalid.

### CODE REVIEW

File: tmcp/message.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def from_dict(cls, message: dict) -> Optional["TMCPMessage"]:
                 direction = action.get("direction", [0.0, 0.0, 0.0])
                 assert isinstance(direction, (list, tuple))
                 assert len(direction) in (2, 3)
+                if len(direction) == 2: direction.append(0.0)
```
Comment (ViliamVadocz) [inline tmcp/message.py]: This won't work if `direction` is a tuple.

Comment (ViliamVadocz) [reply tmcp/message.py]: This would be better:
```py
if len(direction) == 2:
    direction = [direction[0], direction[1], 0.0]
```

**Review by ViliamVadocz — CHANGES_REQUESTED** [review]

File: tmcp/message.py (hunk 1 above)
Comment (VirxEC) [reply tmcp/message.py]: Done

**Comment by ViliamVadocz** [comment]
Thanks for the PR

**Pull request merged**